src/mlpack/bindings/go/print_output_processing.hpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file print_output_processing.hpp
+ * @author Yasmine Dumouchel
+ *
+ * Print the output processing in a Go binding .go file for a given
+ * parameter.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_GO_PRINT_OUTPUT_PROCESSING_HPP
+#define MLPACK_BINDINGS_GO_PRINT_OUTPUT_PROCESSING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "get_type.hpp"
+#include "strip_type.hpp"
+
+namespace mlpack {
+namespace bindings {
+namespace go {
+
+/**
+ * Print output processing for a regular parameter type.
+ */
+template<typename T>
+void PrintOutputProcessing(
+    const util::ParamData& d,
+    const size_t indent,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+    const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type* = 0)
+{
+  const std::string prefix(indent, ' ');
+
+  /**
+   * This gives us code like:
+   *
+   *  <paramName> := GetParam<Type>("paramName")
+   *
+   */
+  std::cout << prefix << d.name << " := GetParam" << GetType<T>(d)
+            << "(\"" << d.name << "\")" << std::endl;
+}
+
+/**
+ * Print output processing for a matrix type.
+ */
+template<typename T>
+void PrintOutputProcessing(
+    const util::ParamData& d,
+    const size_t indent,
+    const typename boost::enable_if<arma::is_arma_type<T>>::type* = 0)
+{
+  const std::string prefix(indent, ' ');
+
+  /**
+   * This gives us code like:
+   *
+   *  var <paramName>Ptr MLPACK_Arma
+   *  <paramName> := <paramName>Ptr.ArmaToGonum_<Type>("paramName")
+   *
+   */
+  std::cout << prefix << "var " << d.name << "Ptr mlpackArma" << std::endl;
+  std::cout << prefix << d.name << " := " << d.name
+            << "Ptr.ArmaToGonum" << GetType<T>(d)
+            << "(\""  << d.name << "\")" << std::endl;
+}
+
+/**
+ * Print output processing for a matrix with info type.
+ */
+template<typename T>
+void PrintOutputProcessing(
+    const util::ParamData& d,
+    const size_t indent,
+    const typename boost::enable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type* = 0)
```

Comment:
One more thing I want to ask here this function is not calling when I `matrix_and_info_out` is enable.
Can you please help me with this.

Reply:
I took a look at the errors you're getting, but I don't see the source code for it.  When you add the option, be sure to use `PARAM_MATRIX_AND_INFO_IN()` to define it.  There are no situations in mlpack where we pass a categorical matrix as output, so I don't think there's a corresponding `PARAM_MATRIX_AND_INFO_OUT()`.  Let me know if I can clarify any further. :+1:

Reply:
I have used the same test_main file that are used python and Julia.

Reply:
Can you make the code available, maybe in a gist or something?  It would be a lot easier to help figure out what's wrong when looking at that.

Reply:
https://pastebin.com/KYk710E3 this is the source file.
https://pastebin.com/uvtDkUFB this is the go output file(at line 261 I have some problem).

Reply:
But when I replace `ArmaToGonumMat` with `ArmaToGonumMatWithInfo` at line number 261 I got this error : 

`[FATAL] Attempted to access parameter --matrix_and_info_out as type St5tupleIJN6mlpack4data13DatasetMapperINS1_15IncrementPolicyENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEN4arma3MatIdEEEE, but its true type is N4arma3MatIdEE!`

Reply:
The `matrix_and_info_out` parameter is a little misleadingly named: it is actually only an `arma::mat` not a categorical matrix.  So you should definitely use ArmaToGonumMat on line 261; that is correct.  As far as I can tell the code you pasted is correct.

Reply:
Sorry about the mistake. I have solved the issue and successfully added `matrix_with_info_in`.
Thank You for the help.